src/ape_ethereum/ecosystem.py

```diff
@@ -154,25 +155,56 @@ def decode_returndata(self, abi: MethodABI, raw_data: bytes) -> Tuple[Any, ...]:
         output_types = [o.canonical_type for o in abi.outputs]  # type: ignore
         try:
             vm_return_values = abi_decode(output_types, raw_data)
-            if not vm_return_values:
-                return vm_return_values
+        except InsufficientDataBytes as err:
+            raise DecodingError() from err
+
+        if not vm_return_values:
+            return vm_return_values
 
-            if not isinstance(vm_return_values, (tuple, list)):
-                vm_return_values = (vm_return_values,)
+        if not isinstance(vm_return_values, (tuple, list)):
+            vm_return_values = (vm_return_values,)
 
-            output_values: List[Any] = []
-            for index in range(len(vm_return_values)):
-                value = vm_return_values[index]
-                if index < len(output_types) and output_types[index] == "address":
+        output_values: List[Any] = []
+        for index in range(len(vm_return_values)):
+            value = vm_return_values[index]
+            if index < len(output_types) and output_types[index] == "address":
+                try:
                     value = self.decode_address(value)
+                except InsufficientDataBytes as err:
+                    raise DecodingError() from err
+
+            output_values.append(value)
+
+        if (
+            len(abi.outputs) == 1
+            and abi.outputs[0].components
+            and all(c.name != "" for c in abi.outputs[0].components)
+        ):
+
+            def create_struct(name: str, types: List[ABIType]):
```

Comment:
Since this is only used the once directly after, we might not need this as a method

Reply:
This might be a good utility function, to have a standard way of describing ABI struct defs